app/packages/osfmap/jsonld.ts

```diff
@@ -0,0 +1,43 @@
+export function *iterOsfmapObjects(osfmapObject: any, propertyPath: string[]): IterableIterator<any> {
+    const [property, ...remainingPath] = propertyPath;
+    const innerObjArray = osfmapObject[property] || [];
+    if (remainingPath.length) {
+        for (const innerObj of innerObjArray) {
+            yield* iterOsfmapObjects(innerObj, remainingPath);
+        }
+    } else {
+        yield* innerObjArray;
+    }
+}
+
+export function *iterOsfmapValues(osfmapObject: any, propertyPath: string[]): IterableIterator<any> {
+    for (const obj of iterOsfmapObjects(osfmapObject, propertyPath)) {
+        yield (Object.hasOwn(obj, '@id') ? obj['@id'] : obj['@value']);
+    }
+}
+
+export function getOsfmapValues(osfmapObject: any, propertyPath: string[]) {
+    return Array.from(iterOsfmapValues(osfmapObject, propertyPath));
+}
+
+export function getSingleOsfmapValue(osfmapObject: any, propertyPath: string[]) {
+    return iterOsfmapValues(osfmapObject, propertyPath).next().value;
+}
```

Comment:
What happens if you use this on something that's not a single value? Does it fail gracefully? Or does typescript help here?

Reply:
It should just get the first one, and if there's no `@id` or `@value`, it should just return `undefined`